### PR TITLE
OME demo server to 5.3

### PR DESCRIPTION
**Pull request by kennethgillen**
Deployment of OME Demo Server on fresh VM, installed almost entirely with Ansible.

Requires some private config, [link for OME team reviewing.](https://github.com/openmicroscopy/management_tools/pull/445)

Tested on a similar vm as follows: 

[ansible_run-ome-demoserver_-infra-testpr_20170612-154806Z.txt](https://github.com/openmicroscopy/infrastructure/files/1068832/ansible_run-ome-demoserver_-infra-testpr_20170612-154806Z.txt)

### CODE REVIEW

**Comment by kennethgillen** [comment]
Note - PR opened on lvm-partition (https://github.com/openmicroscopy/ansible-role-lvm-partition/pull/1) to expose the `shrink` option of `lvol`, which defaults to True. This default was causing idempotency issues, which I worked around in the past by just setting `100%FREE` - whereas now I've fixed it by disabling `shrink`.

**Comment by joshmoore** [comment]
In comparison with the other files in this directory this seems straight-forward enough (though I haven't yet reviewed line by line). However, the duplication between these files is starting to concern me. I've started to unify the various subdirectories into https://github.com/joshmoore/infrastructure/commit/44093e9e27e07d181b468d56780cfe83697c3b2c

@kennethgillen : could you review it and try it out in your local workflow so I know if it's worth continuing?

File: ansible/server-state-playbooks/nightshade-web/playbook.yml [hunk 1]
```diff
@@ -27,11 +27,10 @@
...
       no_log: true
 
-    # This to be refactored for RHEL/non-spacewalk at some point
+    # Now OME are using RHEL without Spacewalk, the current best-method of
+    # checking `is server deployed in Dundee/SLS` is checking for the SLS nameservers.
     - role: openmicroscopy.system-monitor-agent
-      when: >
-            ((ansible_env.COBBLER_SERVER is defined)
-            and (ansible_env.COBBLER_SERVER == "spacewalk.lifesci.dundee.ac.uk"))
+      when: "'10.1.255.216' in ansible_dns.nameservers"
```
Comment (manics) [inline ansible/server-state-playbooks/nightshade-web/playbook.yml]: You could avoid the constant fixes by making it a host (or groupo) var, e.g. `when: "xxx_is_sls_system | default(False)"`

**Comment by kennethgillen** [comment]
@joshmoore  - it's going to take me some time to get my head around what you've done in https://github.com/joshmoore/infrastructure/commit/44093e9e27e07d181b468d56780cfe83697c3b2c

**Comment by joshmoore** [comment]
To explain the steps and/or proposals here:
 
 * playbooks are in `playbook-dir` and named per server
 * the `hosts:` in each playbook are per server
 * all playbooks are listed in `site.yml`
 * all templates are in `playbook-dir/templates`
 * all servers are in `Vagrantfile` (though this could be replaced by molecule)
 * all servers are in `molecule.yml`
 * all tests are under `playbook-dir/tests` (though use of groups would help to make them more generally applicable)
 * there's one requirements.yml for all playbooks in this dir

**Comment by kennethgillen** [comment]
Full run deploying to live server: 
[ansible_run-ome-demoserver-prod-provision-1_20170613-143132Z.txt](https://github.com/openmicroscopy/infrastructure/files/1074790/ansible_run-ome-demoserver-prod-provision-1_20170613-143132Z.txt)

**Comment by kennethgillen** [comment]
Addition of Check_MK logwatch plugin: 
[ansible_run-demoserver_20170622-144101Z.txt](https://github.com/openmicroscopy/infrastructure/files/1095156/ansible_run-demoserver_20170622-144101Z.txt)

**Comment by kennethgillen** [comment]
Addition of check_mk logwatch (with default config) as per ome-demoserver, for ome-dundeeomero.
[ansible_run-monitoring-ome-dundeeomero_20170622-164537Z.txt](https://github.com/openmicroscopy/infrastructure/files/1095541/ansible_run-monitoring-ome-dundeeomero_20170622-164537Z.txt)

**Comment by joshmoore** [comment]
@kennethgillen : before going too much farther, could you look into cherry-pick the above commit here? Otherwise, there are likely to be conflicts.

File: ansible/server-state-playbooks/nightshade-web/playbook.yml (hunk 1 above)
Comment (kennethgillen) [inline ansible/server-state-playbooks/nightshade-web/playbook.yml]: :33 is basically the same as 'is_sls_system' and works 'for free' ie.. without us having to configure anything on the host/group var level. (32,34) were my first crack at that, till we started using RHEL + no-spacewalk.

**Comment by kennethgillen** [comment]
Cherry-picked 🍒

**Comment by joshmoore** [comment]
Cheer(y)s. Once the vault resolution has happened, happy to get this in and look to refactor in further PRs.

**Comment by kennethgillen** [comment]
Ansible Vault removed, replaced with `pass` lookups. MT 445 has had the same treatment, pulling encrypted content from `pass`.

**Comment by rleigh-codelibre** [comment]
Last commit looks fine to me (not tested, just read).

File: ansible/server-state-playbooks/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Server-state playbooks
+----------------------
+
+These playbooks encapsulate the running of various
+production servers run by the OME team. The initial
+template for these playbooks are now contained in:
```
Comment (jburel) [inline ansible/server-state-playbooks/README.md]: typo: ``is now contained``

Comment (kennethgillen) [inline ansible/server-state-playbooks/README.md]: ✅

**Comment by manics** [comment]
Is `molecule test` expected to work?:
```
TASK [openmicroscopy.lvm-partition : storage | create logical volume] **********
task path: /Users/spli/clean/infrastructure/ansible/server-state-playbooks/.molecule/roles/openmicroscopy.lvm-partition/tasks/main.yml:4
fatal: [ns-web.openmicroscopy.org]: FAILED! => {
    "failed": true,
    "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: {{ provision_root_lvname }}: 'provision_root_lvname' is undefined\n\nThe error appears to have been in '/Users/spli/clean/infrastructure/ansible/server-state-playbooks/.molecule/roles/openmicroscopy.lvm-partition/tasks/main.yml': line 4, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: storage | create logical volume\n  ^ here\n"
}

PLAY RECAP *********************************************************************
ns-web.openmicroscopy.org  : ok=2    changed=0    unreachable=0    failed=1
```

**Comment by kennethgillen** [comment]
@manics - not by me at least. I copied it from the original https://github.com/openmicroscopy/ansible-public-omero-example/blob/master/molecule.yml - considered at the time it may be useful, but I've never used molecule myself, so just left it alone. 

Josh's WIP merged in here moved it out of the individual server playbook dirs up a level. 

Would you remove, then, Simon?

**Comment by manics** [comment]
If no-one's using it then remove it?

**Comment by joshmoore** [comment]
LGTM.

**Pull request merged**